src/go/k8s/cmd/main.go

```diff
@@ -361,6 +362,7 @@ func main() {
 			Metrics:                 metricsH,
 			Storage:                 storage,
 			EventRecorder:           helmChartEventRecorder,
+			ControllerName:          "redpanda-helmChart-reconciler",
```

Comment:
The is missing Cache and CacheRecorder with TTL. 

https://github.com/redpanda-data/redpanda-operator/pull/43/files#diff-8659c00d96d07c10eb47505088baad8cb1d6563e41cfc08e532d214816a400efR379-R381